lib/validator.js

```diff
@@ -198,15 +202,65 @@ function validateError(err) {
     return source;
   };
 
+  const isValidMeta = function isValidMeta(metaObj) {
+    if (typeof metaObj !== 'object') {
+      throw new Error("error 'meta' property must be an object");
+    }
+
+    return meta;
+  };
+
+  /**
+   * Determines if the provided number is a valid HTTP status code.
+   *
+   * @param {number} statusCode The status code to validate
+   * @returns {boolean}
+   */
+  const isValidHttpStatusCode = function isValidHttpStatusCode(statusCode) {
+    const validHttpStatusCodes = [
+      // 1XX: Informational
+      100,101, 102, // eslint-disable-line
+
+      // 2XX: Success
+      200, 201, 202, 203, 204, 205, 206, 207, 208, 226, // eslint-disable-line
+
+      // 3XX: Redirection
+      300, 301, 302, 303, 304, 305, 307, 308, // eslint-disable-line
+
+      // 4XX: Client Error
+      400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, // eslint-disable-line
+      413, 414, 415, 416, 417, 418, 421, 422, 423, 424, 426, 428, 429, // eslint-disable-line
+      431, 444, 451, 499, // eslint-disable-line
+
+      // 5XX: Server Error
+      500, 501, 502, 503, 504, 505, 506, 507, 508, 510, 511, 599 // eslint-disable-line
+    ];
+
+    return validHttpStatusCodes.includes(statusCode);
+  };
+
+  const isValidStatus = function isValidStatus(statusCode) {
+    const statusAsNumber = Number(statusCode);
+    if (Number.isNaN(statusAsNumber)) {
+      throw new Error("error 'status' must be a number");
+    }
+
+    if (!isValidHttpStatusCode(statusAsNumber)) {
+      throw new Error("error 'status' must be a valid HTTP status code");
+    }
+
+    return status.toString();
+  };
+
   const error = {};
-  if (id) error.id = id.toString();
-  if (links) error.links = isValidLink(links);
-  if (status) error.status = status.toString();
+  if (id) error.id = id;
+  if (links && Object.keys(links).length) error.links = isValidLink(links);
+  if (status) error.status = isValidStatus(status);
   if (code) error.code = code.toString();
-  if (title) error.title = title.toString();
-  if (detail) error.detail = detail.toString();
-  if (source) error.source = isValidSource(source);
-  if (meta) error.meta = meta;
+  error.title = title ? title.toString() : err.constructor.name;
```

Comment:
I think `err.constructor.name` should be resolved only if a custom `constructor.name` exists.

In case of a simple `Object`, `err.constructor.name` is "Object" and it's not very "pretty" in response.

Could it be:
```js
if (title) {
error.title = title.toString();
} else if (err.constructor.name !== 'Object') {
error.title = err.constructor.name;
}
```

Reply:
Fixed!